query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -323,34 +324,31 @@ private String convertExpressionToMapValueColumn(Expression expression) {
     throw new IllegalArgumentException("operator supports multi value column only");
   }
 
-  private LiteralConstant[] convertExpressionToMapLiterals(Expression expression) {
-    LiteralConstant[] literals = new LiteralConstant[2];
+  private List<LiteralConstant> convertExpressionToMapLiterals(Expression expression) {
+    List<String> literals = new ArrayList<>(List.of("", ""));
     if (expression.getValueCase() == LITERAL) {
       LiteralConstant value = expression.getLiteral();
-      if (value.getValue().getValueType() == ValueType.STRING_ARRAY) {
-        for (int i = 0; i < 2 && i < value.getValue().getStringArrayCount(); i++) {
-          literals[i] =
-              LiteralConstant.newBuilder()
-                  .setValue(
-                      Value.newBuilder().setString(value.getValue().getStringArray(i)).build())
-                  .build();
-        }
+      if (value.getValue().getValueType() == ValueType.STRING_ARRAY
+          && value.getValue().getStringArrayCount() == 2) {
+        literals.set(0, value.getValue().getStringArray(0));
+        literals.set(1, value.getValue().getStringArray(1));
+      } else if (value.getValue().getValueType() == ValueType.STRING) {
```

Comment:
So support for a string arg is the only functional change in this PR, right?

Reply:
Mostly nit changes till now. I thought like and contains_key will not work but with the new changes, they were working so just added unit tests for them till now.

Reply:
resolved